framework/rendering/render_frame.h

```diff
@@ -40,6 +40,12 @@ enum BufferAllocationStrategy
 	MultipleAllocationsPerBuffer
 };
 
+enum DescriptorManagementStrategy
+{
+	StoreInCache,
+	CreateDirectly
```

Comment:
Maybe `Cached` and `Uncached`, instead of `StoreInCache` and `CreateDirectly`?

Reply:
... any maybe this `enum` can even be replaced by a simple `bool`?
Or do you expect to potentially have more than just those two strategies?

Reply:
I chose `enum` over `bool` for 2 reasons:

1. There can potentially be more strategies, for example, cheap frame scope arena allocator managing descriptor sets preallocated from Vulkan (basically don't call `vkResetDescriptorPool` and `vkAllocateDescriptorSets` every frame). 
2. For consistency with `BufferAllocationStrategy` used in the same sample.

For this submission, I can change it to whatever you think is the best.